rpc/handlers.go

```diff
@@ -731,3 +742,48 @@ func setEventFilterRange(filter *blockchain.EventFilter, fromID, toID *BlockID,
 	}
 	return set(blockchain.EventFilterTo, toID)
 }
+
+// AddInvokeTransaction relays an invoke transaction to the gateway.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_write_api.json#L11
+// Note: No checks are performed on the incoming request since we rely on the gateway to perform sanity checks.
+// As this handler is just as a proxy. Any error returned by the gateway is returned to the user as a jsonrpc error.
+func (h *Handler) AddInvokeTransaction(invokeTx json.RawMessage) (*AddInvokeTxResponse, *jsonrpc.Error) {
+	resp, err := h.gatewayClient.AddInvokeTransaction(invokeTx)
+	if err != nil {
+		return nil, &jsonrpc.Error{
+			Code:    getAddInvokeTxCode(err),
+			Message: err.Error(),
+		}
+	}
+
+	invokeRes := new(AddInvokeTxResponse)
+	err = json.Unmarshal(resp, invokeRes)
+	if err != nil {
+		return nil, &jsonrpc.Error{
+			Code:    jsonrpc.InternalError,
+			Message: err.Error(),
+		}
+	}
+
+	return invokeRes, nil
+}
+
+// getAddInvokeTxCode returns the relevant Code for a given addInvokeTx error
+func getAddInvokeTxCode(err error) int {
+	switch {
```

Comment:
@IronGauntlets, I thought you decided it would be best to remove this function, and just return jsonrpc.InvalidParms  (in line 756)?

Reply:
I changed my mind because `h.gatewayClient.AddInvokeTransaction()` can return an error which wraps an HTTP error (see gateway.go:110) which would be incorrect to return as invalid params.